### ISSUE TITLE
issue with frames_include

### ISSUE THREAD

**Issue by Cumol** [MouseLand/suite2p#296]
I wanted to run suite2p on a the first 2000 frames of a dataset that ran fine when using all 8000 frames.

I encountered the following error:

```
registering 2000 frames
Reference frame, 79.42 sec.
2000/2000 frames, 167.99 sec.
bad frames file path: /home/muad/sds-mount/sd18b005/Muad/ca-imaging/exp182-20200211/6drg-gradient-test/RAW/bad_frames.npy

/home/muad/repos/suite2p/suite2p/registration/register.py:113: RuntimeWarning: invalid value encountered in true_divide
  dxy /= dxy.mean()
/home/muad/repos/suite2p/suite2p/registration/register.py:118: RuntimeWarning: invalid value encountered in greater
  ops['badframes'] = np.logical_or(px > ops['th_badframes'] * 100, ops['badframes'])

```

**Comment by carsen-stringer** [comment]
Does your bad_frames file contain frames outside the 2000 frame limit? How are you limiting to 2000 frames?

**Comment by Cumol** [comment]
The bad_frames.npy file does not exist. I always wondered why this line was printed even though there was no file.

I limit to 2000 frames with `frames_include: 2000`

**Comment by nickdelgrosso** [comment]
Fixed by #445

**Comment by Cumol** [comment]
I am having another issue here

The original video has 8001 frames but I only want to run it on the first 2400 and I get the following error:

```
FOUND BINARIES AND OPS IN ['./exp186-20200217/3drg-wt-gradient/RAW/suite2p\\plane0\\ops.npy']
>>>>>>>>>>>>>>>>>>>>> PLANE 0 <<<<<<<<<<<<<<<<<<<<<<
NOTE: not registered / registration forced with ops['do_registration']>1
      (no previous offsets to delete)
----------- REGISTRATION
registering 2400 frames
Reference frame, 69.93 sec.
Registered 2000/2400 in 197.54s
Registered 2400/2400 in 394.46s
Registered 2400/2400 in 591.53s
Registered 2400/2400 in 787.40s

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
C:\Users\DR1F88~1.SIE\AppData\Local\Temp/ipykernel_9448/1342163186.py in <module>
     33 
     34         # run one experiment
---> 35         opsEnd=run_s2p(ops=ops,db=db)

~\Anaconda3\envs\suite2p\lib\site-packages\suite2p\run_s2p.py in run_s2p(ops, db, server)
    446 
    447             print('>>>>>>>>>>>>>>>>>>>>> PLANE %d <<<<<<<<<<<<<<<<<<<<<<'%ipl)
--> 448             op = run_plane(op, ops_path=ops_path)
    449             print('Plane %d processed in %0.2f sec (can open in GUI).' % 
    450                     (ipl, op['timing']['total_plane_runtime']))  

~\Anaconda3\envs\suite2p\lib\site-packages\suite2p\run_s2p.py in run_plane(ops, ops_path, stat)
    200         print('----------- REGISTRATION')
    201         refImg = ops['refImg'] if 'refImg' in ops and ops.get('force_refImg', False) else None
--> 202         ops = registration.register_binary(ops, refImg=refImg) # register binary
    203         np.save(ops['ops_path'], ops)
    204         plane_times['registration'] = time.time()-t11

~\Anaconda3\envs\suite2p\lib\site-packages\suite2p\registration\register.py in register_binary(ops, refImg, raw)
    465 
    466     # return frames which fall outside range
--> 467     ops['badframes'], ops['yrange'], ops['xrange'] = compute_crop(
    468         xoff=ops['xoff'],
    469         yoff=ops['yoff'],

~\Anaconda3\envs\suite2p\lib\site-packages\suite2p\registration\register.py in compute_crop(xoff, yoff, corrXY, th_badframes, badframes, maxregshift, Ly, Lx)
     48     # exclude frames which have a large deviation and/or low correlation
     49     px = dxy / np.maximum(0, cXY)
---> 50     badframes = np.logical_or(px > th_badframes * 100, badframes)
     51     badframes = np.logical_or(abs(xoff) > (maxregshift * Lx * 0.95), badframes)
     52     badframes = np.logical_or(abs(yoff) > (maxregshift * Ly * 0.95), badframes)

ValueError: operands could not be broadcast together with shapes (8001,) (2400,) 

```
I am not sure what is happening exactly but I am guessing that the binary file is created for all 8001 frames instead of only the first 2400, which is why this is happening (but I might be mistaken)
